File: haskell-server/src/BoardGame/Common/Message/SwapPieceResponse.hs

--
-- Copyright 2017 Azad Bolour
-- Licensed under GNU Affero General Public License v3.0 -
--   https://github.com/azadbolour/boardgame/blob/master/LICENSE.md
--

{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE DeriveAnyClass #-}

module BoardGame.Common.Message.SwapPieceResponse (
    SwapPieceResponse(..)
  , tupleToSwapPieceResponse
  ) where

import GHC.Generics (Generic)
import Data.Aeson (FromJSON, ToJSON)
import Control.DeepSeq (NFData)

import BoardGame.Common.Domain.Piece
import BoardGame.Common.Domain.GameMiniState

data SwapPieceResponse = SwapPieceResponse {
    gameMiniState :: GameMiniState
  , piece :: Piece
}
  deriving (Eq, Show, Generic, NFData)

instance FromJSON SwapPieceResponse
instance ToJSON SwapPieceResponse

tupleToSwapPieceResponse :: (GameMiniState, Piece) -> SwapPieceResponse
tupleToSwapPieceResponse (miniState, piece) =
  SwapPieceResponse miniState piece

File: java-client/src/main/java/com/bolour/boardgame/client/message/SwapPieceResponse.java
/*
 * Copyright 2017-2018 Azad Bolour
 * Licensed under GNU Affero General Public License v3.0 -
 *   https://github.com/azadbolour/boardgame/blob/master/LICENSE.md
 */

package com.bolour.boardgame.client.message;

import com.bolour.boardgame.client.domain.GameMiniState;
import com.bolour.boardgame.client.domain.Piece;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class SwapPieceResponse {

    public final GameMiniState gameMiniState;
    public final Piece piece;

    @JsonCreator
    public SwapPieceResponse(
      @JsonProperty("gameMiniState") GameMiniState gameMiniState,
      @JsonProperty("piece") Piece piece) {
        this.gameMiniState = gameMiniState;
        this.piece = piece;
    }
}
